pkg/gadgettracermanager/api/gadgettracermanager.proto

```diff
@@ -95,6 +95,9 @@ message ContainerDefinition {
   uint32 pid = 12;
 
   uint64 netns = 13;
+
+  // Add this field here to be able to print node when using process-collector.
+  string node = 14;
```

Comment:
I don't know, I am not sure we should add the node here, it seems somehow redundant, don't you think?

I am wondering if we should take the opportunity to refactor this gadget and start using the same approach of the socket-collector which composes the JSON on the user side (Using Golang). AFAIU, we are not actually using the advantages of doing it from the eBPF program itself, are we?

If we change the approach, it would be easier to add the node information and any flag we want to add using [this](https://github.com/kinvolk/inspektor-gadget/pull/439/commits/ec27ac53e51dd4f5fadb8385cabd69b1b3447ab3). In addition, we will not need to add the node to either the container-collection or the container eBPF map.

What do you think?

cc @alban @marga-kinvolk @mauriciovasquezbernal

Reply:
I somehow agree with you because when I coded this I found this odd to get the information from golang, write it to the eBPF map, then display it from eBPF code to then read it from golang.
I think we should get from eBPF only the data we cannot get by another way (_e.g._ PID).

Reply:
I agree that handling the json in user side is easier. One possible advantage of generating the json in eBPF is that we could pin this program and the whole data could be accessed with `cat` (or similar), but this is a feature we are not using and I don't think we'll be using it. 

In summary I agree to modify the gadget to fill the k8s related info from user space instead of using the containers map.